Python/bytecodes.c

```diff
@@ -1913,6 +1915,36 @@ dummy_func(
             str = PyStackRef_FromPyObjectSteal(str_o);
         }
 
+        inst(BUILD_INTERPOLATION, (values[4] -- interpolation)) {
```

Comment:
```suggestion
        inst(BUILD_INTERPOLATION, (value, str, format[oparg & 1] -- interpolation)) {
            int conversion = (oparg >> 2);
```

Can you use explicitly named scalar values, using an array is a bit opaque, and error prone as there is no checking that `_PyInterpolation_FromStackRefStealOnSuccess` expects the same number of inputs as the length of `values`.

Since `converison` is only 2 bits of information, it can be put in the `oparg` along with the number of operands.
Many, probably most, interpolations will lack a `format_spec`.